src/helpers/helpers.ts

```diff
@@ -23,4 +26,31 @@ export const formatYear = (year: string): string => {
 export const formatCVC = (cvc: string): string => {
     const cleanedCVC = cvc.replace(/\D+/g, '')
     return cleanedCVC.slice(0, 3);
-}
\ No newline at end of file
+}
+
+export const validateForm = (state: FormData): FormErrors => {
+    
+    const errors: FormErrors = {}
+
+    if (!state[EFormData.CardholderName]) {
+        errors[EFormData.CardholderName] = "Can't be blank";
+    }
+    if (!state[EFormData.CardNumber]) {
+        errors[EFormData.CardNumber] = "Can't be blank";
+    } else if (/[a-zA-Z]/.test(state[EFormData.CardNumber])) {
+        errors[EFormData.CardNumber] = "Wrong format, numbers only";
+    } else if (!/^\d{16}$/.test(state[EFormData.CardNumber].replace(/\s/g, ''))) {
+        errors[EFormData.CardNumber] = "Must be 16 digits";
+    }
+    if (!state[EFormData.ExpMonth]) {
+        errors[EFormData.ExpMonth] = "Can't be blank";
+    }
+    if (!state[EFormData.ExpYear]) {
+        errors[EFormData.ExpYear] = "Can't be blank";
+    }
+    if (!state[EFormData.CVC]) {
```

Comment:
this CVC validation should check if the input is 3 digits cos CVC are made of 3 digits always.

https://www.loom.com/share/21123672a1b24272a48c52a44e8ff431